Code/Editor/LogFile.cpp

```diff
@@ -178,6 +181,140 @@ void CLogFile::FormatLineV(const char * format, va_list argList)
+                        hResult = classObjectEnumerator->Next(WBEM_INFINITE, 1, &classObject, &numResults);
+
+                        if (!FAILED(hResult) && classObject && numResults != 0)
+                        {
+                            constexpr const wchar_t* namePropertyName{ L"Name" };
+                            constexpr const wchar_t* versionPropertyName{ L"Version" };
+                            // get the class object's property value
+                            VARIANT propertyValue;
+                            // Query the Name field
+                            if (!FAILED(classObject->Get(namePropertyName, 0, &propertyValue, nullptr, nullptr)))
+                            {
+                                // If the name is a binary string copy it over
+                                if ((propertyValue.vt & VT_BSTR) == VT_BSTR)
+                                {
+                                    AZStd::to_string(osInfo.m_name, V_BSTR(&propertyValue));
+                                }
+                            }
+                            // VariantClear must be called on the variant retrieved from `IWbemClassObject::Get`
+                            // according to the documentation:
+                            // https://learn.microsoft.com/en-us/windows/win32/api/wbemcli/nf-wbemcli-iwbemclassobject-get
+                            VariantClear(&propertyValue);
+
+                            // Query the Version field
+                            if (!FAILED(classObject->Get(versionPropertyName, 0, &propertyValue, nullptr, nullptr)))
+                            {
+                                // If the name is a binary string copy it over
+                                if ((propertyValue.vt & VT_BSTR) == VT_BSTR)
+                                {
+                                    AZStd::to_string(osInfo.m_version, V_BSTR(&propertyValue));
+                                }
+                            }
+                            // VariantClear must be called on the variant retrieved from `IWbemClassObject::Get`
+                            // according to the documentation:
+                            // https://learn.microsoft.com/en-us/windows/win32/api/wbemcli/nf-wbemcli-iwbemclassobject-get
+                            VariantClear(&propertyValue);
+                        }
+
+                        if (classObject)
+                        {
+                            classObject->Release();
+                        }
+                    }
+
+                    if (classObjectEnumerator)
+                    {
+                        classObjectEnumerator->Release();
+                    }
+                }
+            }
+
+            if (services)
+            {
+                services->Release();
+            }
+        }
+
+        if (locator)
+        {
+            locator->Release();
+        }
+
+        CoUninitialize();
+
+        return osInfo;
```

Comment:
There is no warning or indication of any of this function failing other than returning an uninitialized OSInfo. Maybe at least provide a warning?

Reply:
I'll look into what we should do in the case of an error.
If it fails we could fall back to using GetVersionEx.
Your thoughts @spham-amzn

Reply:
I think that is a reasonable fallback

Reply:
Done.